### ISSUE TITLE
Pulling the OCIR images to OKE without Secrets on OKE Private subnet(No NAT gateway)

### ISSUE THREAD

**Issue by arunreddy00** [oracle-devrel/oke-credential-provider-for-ocir#3]
Hi team,

I have No internet access to the worker node.

I have followed all the steps & pre-requites of IAM for instance principals. However, I'm not able to pull images from the OCI registry.
Please help me on how to troubleshoot & pull the images from the private registry using credentials provider.

`
  Warning  Failed     8m13s (x4 over 9m48s)   kubelet            Failed to pull image "iad.ocir.io/namespace/image:tag": initializing source docker://iad.ocir.io/namespace/image:tag: reading manifest node-v3.28.2 in iad.ocir.io/namespace/image: denied: Anonymous users are only allowed read access on public repos
  `

Questions:
- Is it achievable to pull images on worker node running in private subnet ?

**Comment by ivandelic** [comment]
Yes, pulling from a private subnet is possible. You need a Service Gateway and the proper route within the worker subnet. 

The error message (`denied: Anonymous users are only allowed read access on public repos`) is more about missing policies.
Did you set up the IAM rules (explained [here](https://github.com/oracle-devrel/oke-credential-provider-for-ocir/tree/main/examples/run-on-worker-instance-principal)) properly?

**Comment by joelp172** [comment]
I have been facing a similar issue. Double check that the registry address has been included in the `credential-provider-config.yaml` I had to append my domains to the matchImages config, as explained [here](https://kubernetes.io/docs/tasks/administer-cluster/kubelet-credential-provider/#configure-image-matching).

Or it could be this [issue](https://github.com/oracle-devrel/oke-credential-provider-for-ocir/issues/6).
